### ISSUE TITLE
[clangd][include-cleaner] False warning: Included header X is not used directly

### ISSUE THREAD

**Issue by tuero** [llvm/llvm-project#75777]
```shell
$ ./clangd --version
clangd version 17.0.3 (https://github.com/llvm/llvm-project 888437e1b60011b8a375dd30928ec925b448da57)
Features: linux+grpc
Platform: x86_64-unknown-linux-gnu
```

If I have a header file which defines a concept, and I include that header file and reuse that concept, I am getting a false `Included header X is not used directly`, i.e., by commenting out that header I get a compile error (because that concept was not previously defined.

I have a reproducible repo here, but will also type out below: https://github.com/tuero/clangd_include_example

util.h
```cpp
#ifndef UTIL_H_
#define UTIL_H_

#include <string>

// Concept for simple states for flat search
template <typename T>
concept SimpleEnv = requires(T t, const T ct, const std::string &s) {
    T(s);
    { ct.is_terminal() } -> std::same_as<bool>;
};

#endif    // UTIL_H_
```

main.cpp
```cpp
#include <iostream>
#include <string>

#include "util.h"   // Included header util.h is not used directly

struct X {
    X(const std::string &s) : s(s) {}
    bool is_terminal() const {
        return false;
    }
    std::string s;
};

template <typename T>
concept Env = SimpleEnv<T>;    // Here we use the concept defined in util.h, 
                                                    // commenting header out will cause compiler error here

template <Env T>
bool is_terminal(const T &t) {
    return t.is_terminal();
}

int main() {
    X x("hello");
    std::cout << is_terminal(x) << std::endl;
}
```

**Comment by HighCommander4** [comment]
Duplicate of #60702?

**Comment by tuero** [comment]
That does indeed seem like a duplicate. Would that commit be included in `17.0.3`? I cant tell from running `git tag --contains`; the commit was on Aug 31, and `17.0.3` released on Oc17. I have not tried doing a manual build at head to confirm.

**Comment by HighCommander4** [comment]
> Would that commit be included in `17.0.3`?

No, it is not. The commit is only on trunk, and 17.x was branched sometime in July. The fix will appear in clangd 18.

If you'd like a prebuilt version of clangd that contains the fix, we have weekly snapshots of trunk at https://github.com/clangd/clangd/releases.

**Comment by HighCommander4** [comment]
I tested your example with trunk, and it is indeed fixed there.

**Comment by HighCommander4** [comment]
Duplicate of https://github.com/llvm/llvm-project/issues/60702

**Comment by tuero** [comment]
Ah thanks for the info, I can also see its confirmed on my end using the snapshot!
